### PR TITLE
Add collectAll method to the Meter and MeterProvider

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/MeterSdk.java [hunk 1]
```diff
@@ -81,4 +84,14 @@ public LabelSetSdk createLabelSet(String... keyValuePairs) {
   public LabelSetSdk createLabelSet(Map<String, String> labels) {
     return LabelSetSdk.create(labels);
   }
+
+  Collection<MetricData> collectAll() {
+    InstrumentRegistry instrumentRegistry = meterSharedState.getInstrumentRegistry();
+    Collection<AbstractInstrument> instruments = instrumentRegistry.getInstruments();
+    ArrayList<MetricData> ret = new ArrayList<>(instruments.size());
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/MeterSdk.java]: left hand side should just be declared as `List<MetricData>`
I'd also prefer the temporary variable be called `results`, but that's a nitpick.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/MeterSdkProvider.java [hunk 2]
```diff
@@ -50,6 +54,20 @@ public MeterSdk get(String instrumentationName, String instrumentationVersion) {
...
   }
 
+  /**
+   * Collects all the metrics and returns a collection of collected {@link MetricData}.
+   *
+   * @return a collection of collected {@link MetricData}.
+   */
+  public Collection<MetricData> collectAll() {
+    Collection<MeterSdk> meters = registry.getComponents();
+    ArrayList<MetricData> ret = new ArrayList<>(meters.size());
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/MeterSdkProvider.java]: `List<MetricData>` on the LHS

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/MeterSdk.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/MeterSdk.java]: Done.

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/MeterSdkProvider.java (hunk 2 above)
Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/metrics/MeterSdkProvider.java]: Done.

**Pull request merged**